### PR TITLE
Add support for custom package generated with composer archive

### LINKED ISSUE #26: Hosted repository: the dependencies are not set from the uploaded archive
* **What are you trying to do?**
I've uploaded a ZIP archive using `composer archive --format=zip` then `curl -v --user '...' --upload-file ...` as documented here: https://github.com/sonatype-nexus-community/nexus-repository-composer/blob/master/docs/COMPOSER_USER_DOCUMENTATION.md#user-content-publishing-composer-packages

The issue is that my package has dependencies listed in the `require` part of its `composer.json` file. Those dependencies are not present in the plugin's generated JSON.
The consequence is that I can't require my module in an other project without having to explicitly require all of its dependencies.

* **What feature or behavior is this required for?**
Having private hosted packages.

* **How could we solve this issue? (Not knowing is okay!)**
I think we just have to add the `require` part of the archive's `composer.json` file into the generated json.

* **Anything else?**
Nope, thank you for your work !

### PR DESCRIPTION
**Pull request by Elendev**
The command `composer archive --format=zip` generate a ZIP archive with the `composer.json` file at the root of the archive.
The nexus-repository-composer is unable to handle this kind of archives correctly since it search for `composer.json` inside a subdirectory of the given archive.
This PR modify the way the `composer.json` is fetched in the ZIP archive to allow to handle the archives generates by `composer`.

This pull request makes the following changes:
* Find the `composer.json` file even if it's on the root of the archive
* Add corresponding unit tests

It relates to the following issue :
* Fixes #26

### CODE REVIEW

**Comment by fjmilens3** [comment]
Thanks for putting this together, @Elendev! I don't have time to take a look at it _right_ at the moment but we'll make sure we do.

Beyond that, I would suggest that you add your name to the contributors file as part of this PR! :)

**Comment by Elendev** [comment]
@fjmilens3 done, I've added myself to the contributor list :-)

**Comment by HabibMAALEM** [comment]
Hi guys, @fjmilens3 can you approve this PR to finish the merge

File: src/test/resources/org/sonatype/nexus/repository/composer/internal/extractInfoFromZipballWithJsonComposerArchived.composer.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{
+    "name": "vendor/project",
+	"version" : "1.2.3",
```
Comment (fjmilens3) [inline src/test/resources/org/sonatype/nexus/repository/composer/internal/extractInfoFromZipballWithJsonComposerArchived.composer.json]: Minor: Spacing (doesn't line up with the other JSON formatting).

File: src/test/resources/org/sonatype/nexus/repository/composer/internal/extractInfoFromZipballWithJsonComposerArchived.composer.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    "description": "Test description",
+    "keywords": ["keyword1", "keyword2", "keyword3"],
+    "homepage": "http://www.example.com/",
+	"time": "2008-05-15",
```
Comment (fjmilens3) [inline src/test/resources/org/sonatype/nexus/repository/composer/internal/extractInfoFromZipballWithJsonComposerArchived.composer.json]: Minor: Spacing (doesn't line up with the other JSON formatting).

File: src/test/resources/org/sonatype/nexus/repository/composer/internal/extractInfoFromZipballWithJsonComposerArchived.composer.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            "homepage": "http://www.example.com/author1"
+        },
+        {
+            "name": "Author2",
+            "email": "author2@example.com",
+            "homepage": "http://www.example.com/author2"
+        }
+    ],
+    "support": {
+		"email": "email@example.com",
```
Comment (fjmilens3) [inline src/test/resources/org/sonatype/nexus/repository/composer/internal/extractInfoFromZipballWithJsonComposerArchived.composer.json]: Minor: Spacing (doesn't line up with the other JSON formatting).

File: src/test/java/org/sonatype/nexus/repository/composer/internal/ComposerJsonExtractorTest.java [hunk 4]
```diff
@@ -58,6 +58,20 @@ public void extractInfoFromZipballWithJson() throws Exception {
...
   }
 
+  @Test
+  public void extractInfoFromZipballWithJsonComposerArchived() throws Exception {
+    String expected;
+    try (InputStream in = getClass().getResourceAsStream("extractInfoFromZipballWithJsonComposerArchived.composer.json")) {
+      expected = CharStreams.toString(new InputStreamReader(in, StandardCharsets.UTF_8));
+    }
+    String actual;
+    try (InputStream in = getClass().getResourceAsStream("extractInfoFromTipBallWithJsonComposerArchived.zip")) {
```
Comment (fjmilens3) [inline src/test/java/org/sonatype/nexus/repository/composer/internal/ComposerJsonExtractorTest.java]: Minor: Change "TipBall" to "Zipball" here and in the associated file. I'm guessing one's just an accident from the T and the Z being side by side on QWERTZ layouts, and the other change is just to be consistent with our naming conventions elsewhere.

**Comment by fjmilens3** [comment]
@Elendev, made a few very minor (and pedantic) code style and formatting notes. Once those are changed I'm comfortable with merging this down, thanks for putting it together.

@HabibMAALEM, see the above comments, we should be able to get this merged in shortly.

@TheBay0r, if you're interested, you may want to give this a quick test yourself as our unofficial QA expert as well. I don't think it's exactly the riskiest change in the world, however!

**Comment by TheBay0r** [comment]
@fjmilens3 I'll have a look either today or Monday 👍

**Comment by TheBay0r** [comment]
I just had a look. I was able to reproduce the initial issue and it seems it is fixe with this branch. Also regression seems fine!
Looks good from my point of view 👍

**Comment by Elendev** [comment]
@fjmilens3 I've fixed the typo and the formatting issues

**Pull request merged**